crates/build-utils/src/filter.rs

```diff
@@ -0,0 +1,176 @@
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use std::{collections::BTreeMap, fs};
+
+use crate::path::PathWrapper;
+
+type Folder = String;
+type FilterMap = BTreeMap<Folder, Vec<String>>;
+
+/// Filter to be applied on the tests files
+#[derive(Deserialize, Default, Serialize)]
+pub struct Filter {
+    /// Mapping containing the directories and the files that should be skipped
+    filename: FilterMap,
+    /// Mapping containing the directories and the regex patterns that should be skipped
+    regex: FilterMap,
+    /// Mapping containing the directories and the specific tests that should be skipped
+    #[serde(rename = "testname")]
+    test_name: FilterMap,
+}
+
+impl Filter {
+    pub fn load_file(path: &str) -> Result<Self, eyre::Error> {
+        let filter = fs::read_to_string(path)?;
+        Ok(serde_yaml::from_str(&filter)?)
+    }
+
+    /// Checks if the given path is inside the filter object
+    pub fn is_skipped(&self, path: &PathWrapper, case_name: Option<String>) -> bool {
+        let dir_name = path.parent().file_stem_to_string();
+        let file_name = path.file_stem_to_string();
+
+        let mut should_skip = self
+            .filename
+            .get(&dir_name)
+            .map(|filtered_files| filtered_files.iter().any(|filename| filename == &file_name))
+            .unwrap_or_default();
+
+        should_skip |= self
+            .regex
+            .get(&dir_name)
+            .map(|regexes| {
+                regexes.iter().any(|regex| {
+                    Regex::new(regex.as_str())
+                        .expect("Error with regex pattern")
+                        .is_match(&file_name)
+                })
+            })
+            .unwrap_or_default();
+
+        if let Some(case_name) = case_name {
+            should_skip |= self
+                .test_name
+                .get(&dir_name)
+                .map(|tests| tests.iter().any(|test| test == &case_name))
+                .unwrap_or_default();
+        }
+
+        should_skip
+    }
+
+    /// Returns the difference in keys (folders) between the two filters
+    pub fn diff(&self, rhs: &Self) -> Vec<Folder> {
+        let mut diff = Vec::new();
+        diff.append(&mut map_diff(&self.filename, &rhs.filename));
+        diff.append(&mut map_diff(&self.regex, &rhs.regex));
+        diff.append(&mut map_diff(&self.test_name, &rhs.test_name));
+        diff
+    }
+}
+
+fn map_diff(lhs: &FilterMap, rhs: &FilterMap) -> Vec<Folder> {
```

Comment:
what is lhs and rhs?

Reply:
left hand side and right hand side

Reply:
hahaha I have no clue what latin hypercube sampling is, but lhs and rhs is quite common in rust when doing [operations](https://doc.rust-lang.org/std/ops/trait.Mul.html).